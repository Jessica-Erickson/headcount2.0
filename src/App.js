import React, { Component } from 'react';
import './App.css';
import DistrictRepository from './helper';
import kinderData from './data/kindergartners_in_full_day_program';
import onlineEnrollment from './data/online_pupil_enrollment';
import titleIStudents from './data/title_i_students';
import thirdGradeTests from './data/3rd_grade_tests';
import eighthGradeTests from './data/8th_grade_test_scores';
import aveEthMath from './data/average_race_ethnicity_math_scores';
import aveEthReading from './data/average_race_ethnicity_reading_scores';
import aveEthWriting from './data/average_race_ethnicity_writing_scores';
import dropoutRates from './data/dropout_rates_by_race_and_ethnicity';
import hsGraduation from './data/high_school_graduation_rates';
import medianIncome from './data/median_household_income';
import regEnrollment from './data/pupil_enrollment';
import enrollmentByEth from './data/pupil_enrollment_by_race_ethnicity';
import remediation from './data/remediation_in_higher_education';
import childrenInPoverty from './data/school_aged_children_in_poverty';
import specialEd from './data/special_education';
import frLunch from './data/students_qualifying_for_free_or_reduced_price_lunch';
import Controls from './Controls';
import Compare from './Compare';

export default class App extends Component {
  constructor() {
    super()
    this.state = {
      search: '',
      compare: [],
      repo: new DistrictRepository(kinderData)
    }

    this.data = {
      'Kindergartners In Full Day Program': kinderData,
      'Online Pupil Enrollment': onlineEnrollment,
      'Title I Students': titleIStudents,
      '3rd Grade Tests': thirdGradeTests,
      '8th Grade Test Scores': eighthGradeTests,
      'Average Ethnicity Math Scores': aveEthMath,
      'Average Ethnicity Reading Scores': aveEthReading,
      'Average Ethnicity Writing Scores': aveEthWriting,
      'Dropout Rates by Ethnicity': dropoutRates,
      'High School Graduation Rates': hsGraduation,
      'Median Household Income': medianIncome,
      'Pupil Enrollment': regEnrollment,
      'Pupil Enrollment by Ethnicity': enrollmentByEth,
      'Remediation in Higher Education': remediation,
      'School Aged Children in Poverty': childrenInPoverty,
      'Special Education': specialEd,
      'Students Qualifying for Free or Reduced Price Lunch': frLunch
    }
  }

  handleSearch = (event) => {
    this.setState({search: event.target.value});
  }

  handleCardClick = (name) => {
    if (this.state.compare.length < 2) {
      this.setState({compare: [...this.state.compare, name]}); 
    }
  }

  handleCompareClick = (name) => {
    this.setState({
      compare: 
      this.state.compare.filter(card => {
        return card !== name;
      })
    });
  }

  handleHeaderClick = (name) => {
    this.setState({ 
      repo: new DistrictRepository(this.data[name])
    });
  }

  render() {
    return (
      <div className='App'>
        <Controls 
          options={Object.keys(this.data)} 
          handleHeaderClick={this.handleHeaderClick} />
        <Compare 
          toCompare={this.state.compare}
          repo={this.state.repo}
          handleCompareClick={this.handleCompareClick} /> 
      </div>
    );
  }
}
