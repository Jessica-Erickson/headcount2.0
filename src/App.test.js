import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {shallow} from 'enzyme';
import DistrictRepository from './helper';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot()
  });

  it('should keep track of what is in the search bar', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.state('search')).toEqual('');
  });

  it('should keep track of which cards to compare', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.state('compare')).toEqual([]);
  });

  it('should keep a repo of the relevant data', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.state('repo')).toBeInstanceOf(DistrictRepository);
  });

  it('should update search when handleSearch is called', () => {
    const wrapper = shallow(<App />);
    const testEvent = {target: {value: 'denver'}};
    const expected = 'denver';

    wrapper.instance().handleSearch(testEvent);

    expect(wrapper.state('search')).toEqual(expected);
  });

  it('should update compare when a card is clicked', () => {
    const wrapper = shallow(<App />);
    const name = 'AGATE 300';
    const expected = ['AGATE 300'];

    wrapper.instance().handleCardClick(name);

    expect(wrapper.state('compare')).toEqual(expected);
  });

  it('should add up to two cards in compare', () => {
    const wrapper = shallow(<App />);
    const name1 = 'AGATE 300';
    const name2 = 'MOFFAT 2';
    const name3 = 'LONE STAR 101';
    const expected = ['AGATE 300', 'MOFFAT 2'];

    wrapper.instance().handleCardClick(name1);
    wrapper.instance().handleCardClick(name2);
    wrapper.instance().handleCardClick(name3);

    expect(wrapper.state('compare')).toEqual(expected);
  });

  it('should remove compare cards when clicked', () => {
    const wrapper = shallow(<App />);
    const name1 = 'AGATE 300';
    const name2 = 'MOFFAT 2';
    const expected = ['MOFFAT 2'];

    wrapper.instance().handleCardClick(name1);
    wrapper.instance().handleCardClick(name2);

    wrapper.instance().handleCompareClick(name1);

    expect(wrapper.state('compare')).toEqual(expected);
  });

  it('should be able to add new cards for comparison', () => {
    const wrapper = shallow(<App />);
    const name1 = 'AGATE 300';
    const name2 = 'MOFFAT 2';
    const name3 = 'LONE STAR 101';
    const expected = ['AGATE 300', 'MOFFAT 2'];

    wrapper.instance().handleCardClick(name3);
    wrapper.instance().handleCardClick(name1);

    wrapper.instance().handleCompareClick(name3);

    wrapper.instance().handleCardClick(name2);

    expect(wrapper.state('compare')).toEqual(expected);
  });
});
