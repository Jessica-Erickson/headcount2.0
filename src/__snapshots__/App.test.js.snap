// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Controls
    handleHeaderClick={[Function]}
    options={
      Array [
        "Kindergartners In Full Day Program",
        "Online Pupil Enrollment",
        "Title I Students",
        "3rd Grade Tests",
        "8th Grade Test Scores",
        "Average Ethnicity Math Scores",
        "Average Ethnicity Reading Scores",
        "Average Ethnicity Writing Scores",
        "Dropout Rates by Ethnicity",
        "High School Graduation Rates",
        "Median Household Income",
        "Pupil Enrollment",
        "Pupil Enrollment by Ethnicity",
        "Remediation in Higher Education",
        "School Aged Children in Poverty",
        "Special Education",
        "Students Qualifying for Free or Reduced Price Lunch",
      ]
    }
  />
</div>
`;
